towerlib/towerlib.py

```diff
@@ -1997,24 +2034,707 @@ def schedules(self):
+                playbooks = project.playbooks
+                break
+        return playbooks
+
+    def get_production_branch_playbooks(self, scm_url):
+        """Get all the playbooks of the project, which has the branch production as scm branch.
+
+        Args:
+            scm_url: given scm url
+
+        Returns:
+            list: a list with the names of all the playbooks of production branch project.
+
+        """
+        projects = self.get_all_projects()
+        for project in projects:
+            if project.scm_url == scm_url and project.scm_branch == 'production':
+                playbooks = project.playbooks
+                return playbooks
+
+    def get_ansible_facts_by_host_id(self, host_id):
+        """Get the ansible_facts of the given host.
+
+        Args:
+            host_id: id of the host for which the method will return the ansible_facts.
+
+        Returns:
+            dict: ansible_facts as dictionary.
+
+        """
+        host = self.get_host_by_id(host_id)
+        return host.ansible_facts()
+
+    def get_all_inventories(self):
+        """
+        Get all the inventories of the ansible tower.
+
+        Returns:
+            list: list of inventories as EntityManager object.
+        """
+        inventories = [item for item in self.inventories]
+        return inventories
+
+    def get_all_hosts(self):
+        """Get all the hosts of the given ansible tower.
+
+        Returns:
+            list: list of all the hosts as EntityManager object.
+
+        """
+        hosts = [item for item in self.hosts]
+        return hosts
+
+    def get_all_credentials(self):
+        """Get all the credentials of the ansible tower.
+
+        Returns:
+            list: the list of credentials as EntityManager object.
+
+        """
+        credentials = [item for item in self.credentials]
+        return credentials
+
+    def get_hosts_by_inventory_id(self, inventory_id):
+        """Get filtered list of hosts for a given inventory id.
+
+        Args:
+             inventory_id: the given inventory id.
+
+        Returns:
+            list: the list of filtered host as EntityManger object.
+
+        """
+        hosts = [item for item in self.hosts if item.inventory.id == inventory_id]
+        return hosts
+
+    def get_all_jobs(self):
+        """Get all the jobs of the ansible tower.
+
+        Returns:
+            list: list of all the jobs as EntityManager object
+
+        """
+        jobs = [item for item in self.jobs]
+        return jobs
+
+    def get_jobs_by_name(self, given_job_name):
+        """Get filtered list of jobs for a given name.
+
+        Args:
+            given_job_name: the given job name.
+
+        Returns:
+             list: the filtered list of jobs.
+
+        """
+        job_list = [item for item in self.jobs if item.name == given_job_name]
+        return job_list
+
+    def get_project_updates(self):
```

Comment:
Please model the project_update as an entity to be returned by the EntityManager to be compatible with the project's design principles.

Reply:
Implemented.